cmd/branchingconfigmanagers/release-controller-config-manager/main.go

```diff
@@ -0,0 +1,148 @@
+	howManyDaysBefore                  = 14
+	lifecycleConfigFilePathDefaultPath = ""
+)
+
+type options struct {
+	curOCPVersion               string
+	futureOCPVersion            string
+	releaseControllerConfigsDir string
+	releaseRepoDir              string
+	outDir                      string
+	lifecycleConfigFilePath     string
+	logLevel                    int
+}
+
+func (o *options) overrideOCPVersions(curVersion, futureVersion string) {
+	if o.curOCPVersion == "" {
+		o.curOCPVersion = curVersion
+	}
+	if o.futureOCPVersion == "" {
+		o.futureOCPVersion = futureVersion
+	}
+}
+
+func (o *options) adjustPaths() error {
+	if o.releaseRepoDir != "" {
+		if !path.IsAbs(o.releaseRepoDir) {
+			return errors.New("release-repo-path path has to be absolute")
+		}
+
+		if path.IsAbs(o.releaseControllerConfigsDir) {
+			return errors.New("release-controllers path can't be absolute when release-controllers is specified")
+		}
+
+		o.releaseControllerConfigsDir = path.Join(o.releaseRepoDir, o.releaseControllerConfigsDir)
+	}
+	return nil
+}
+
+func gatherOptions() (*options, error) {
+	var errs []error
+	o := &options{}
+	flag.StringVar(&o.curOCPVersion, "current-release", "", "Current OCP version")
+	flag.StringVar(&o.futureOCPVersion, "future-release", "", "Next OCP version")
+	flag.StringVar(&o.releaseControllerConfigsDir, "release-controllers", releaseControllerConfigPath, "Path to release controllers config .json files")
+	flag.StringVar(&o.releaseRepoDir, "release-repo-path", "", "Path to 'openshift/release/ folder")
+	flag.StringVar(&o.outDir, "out", "", "Output directory. Default to --release-controllers dir")
+	flag.StringVar(&o.lifecycleConfigFilePath, "lifecycle-config", lifecycleConfigFilePathDefaultPath, "Path to the lifecycle config file")
+	flag.IntVar(&o.logLevel, "log-level", int(logrus.DebugLevel), "Log level")
+	flag.Parse()
+
+	if _, err := ocplifecycle.ParseMajorMinor(o.curOCPVersion); o.curOCPVersion != "" && err != nil {
+		errs = append(errs, fmt.Errorf("error parsing current-release %s", o.curOCPVersion))
+	}
+
+	if _, err := ocplifecycle.ParseMajorMinor(o.futureOCPVersion); o.futureOCPVersion != "" && err != nil {
+		errs = append(errs, fmt.Errorf("error parsing future-release %s", o.futureOCPVersion))
+	}
+
+	if o.lifecycleConfigFilePath == "" {
+		errs = append(errs, errors.New("lifecycle-config is mandatory"))
+	}
+	if !path.IsAbs(o.lifecycleConfigFilePath) {
+		errs = append(errs, errors.New("lifecycle-config path is not absolute"))
+	}
+
+	if o.outDir != "" && !path.IsAbs(o.outDir) {
+		errs = append(errs, errors.New("out path is not absolute"))
+	}
+
+	return o, utilerrors.NewAggregate(errs)
+}
+
+func main() {
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Error("failed to gather options")
+		os.Exit(optsParsingErrorExitStatus)
+	}
+
+	if err = o.adjustPaths(); err != nil {
+		logrus.WithError(err).Error("failed to gather options")
+		os.Exit(optsParsingErrorExitStatus)
```

Comment:
```suggestion
		logrus.WithError(err).Fatal("failed to adjust paths")
```
You can remove the `os.Exit`